### DISCUSSION TITLE
A 'AdvectionBC' is not a registered object.

### DISCUSSION THREAD

**Discussion by hhy2022** [idaholab/moose#21401] [Q&A General] [answered]
Hi, I am trying to develop my own app(name Myself) using MOOSE. I found one bc object called "AdvectionBC" under [(modules/navier_stokes] module (https://mooseframework.inl.gov/source/bcs/AdvectionBC.html#fef0c6df-6c16-4af7-a474-58cc8577c10a). So I copied this one AdvectionBC.h & AdvectionBC.C and changed the register code in AdvectionBC.C file:

`registerMooseObject("MyselfAPP", AdvectionBC);`

This is the only line I changed to the source code. I put the .h file in the `include` file and .c file in the `src` file. I could compile the app but when I am trying to running the input file, it gave me a message that `A 'AdvectionBC' is not a registered object.`.

I am kind of confused since I thought I had registered this object. Is there any other reason that make the registration of object fail?

Thank you so much for any kind of help.

**Comment by GiudGiud** [comment, marked as answer]
Hello

So first, you dont need to move code over to use these objects. You just need to enable NAVIER_STOKES in the makefile of your application.

Secondly, is your called MyselfApp or myselfApp ? The case is important. you can check the name in src/base/MyselfApp.C.

Guillaume

**Reply by hhy2022** [reply]
Yes. I understood I can use the module by enabling it. I just want to build my own application and this is the necessary part.
My application is called MyselfAPP. I saw the "MyselfAPP.C" file in my src file.

**Reply by GiudGiud** [reply]
then it needs to be registered to MyselfAPP not MyselfApp but that seems weird to me
Just be careful with the case.
If you paste the myselfApp.C file here I can check the app name

**Reply by hhy2022** [reply]
I see. I didn't even notice about the case difference. Thank you so much. I changed the name to make sure it was exactly the same as my application name and it worked well now.

Thank you so much for your help.
